lib/active_model_serializers/test/schema.rb

```diff
@@ -10,19 +10,36 @@ module Schema
       #   get :index
       #   assert_response_schema
       def assert_response_schema(schema_path = nil, message = nil)
-        matcher = AssertResponseSchema.new(schema_path, response, message)
+        matcher = AssertResponseSchema.new(schema_path, request, response, message)
+        assert(matcher.call, matcher.message)
+      end
+
+      def assert_request_schema(schema_path = nil, message = nil)
+        matcher = AssertRequestSchema.new(schema_path, request, response, message)
+        assert(matcher.call, matcher.message)
+      end
+
+      def assert_request_response_schema(schema_path = nil, message = nil)
+        assert_request_schema(schema_path, message)
+        assert_response_schema(schema_path, message)
+      end
+
+      def assert_schema(payload, schema_path = nil, message = nil)
+        matcher = AssertSchema.new(schema_path, request, response, message, payload)
         assert(matcher.call, matcher.message)
       end
 
       MissingSchema = Class.new(Errno::ENOENT)
       InvalidSchemaError = Class.new(StandardError)
 
-      class AssertResponseSchema
-        attr_reader :schema_path, :response, :message
+      class AssertSchema
+        attr_reader :schema_path, :request, :response, :message, :payload
 
-        def initialize(schema_path, response, message)
```

Comment:
- we always need a payload
- we always need the request if schema_path is nil
  -  GET, DELETE we don't need the response_body
  - else PUT, POST, PATCH we do need the response body

so a generic assertion would just need the `schema_path, message, and payload`

the request assertion would also need the `request`

and  the response assertion would also need the `request, and response`.. so I guess we can do attribute overloading here...